cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -0,0 +1,380 @@
+        conn.setRequestProperty(AUTHORIZATION, "Bearer " + this.accessToken);
+        conn.setRequestProperty(PUSH_ORGANIZATION, org);
+        conn.setRequestProperty(CONTENT_TYPE, APPLICATION_OCTET_STREAM);
+
+        conn.setDoOutput(true);
+        conn.setChunkedStreamingMode(BUFFER_SIZE);
+
+        try (DataOutputStream outputStream = new DataOutputStream(conn.getOutputStream())) {
+            // Send balo content by 1 kb chunks
+            byte[] buffer = new byte[BUFFER_SIZE];
+            int count;
+            try (ProgressBar progressBar = new ProgressBar(
+                    org + "/" + name + ":" + version + " [project repo -> central]", getTotalFileSizeInKB(baloPath),
+                    1000, outStream, ProgressBarStyle.ASCII, " KB", 1);
+                    FileInputStream fis = new FileInputStream(baloPath.toFile())) {
+                while ((count = fis.read(buffer)) > 0) {
+                    outputStream.write(buffer, 0, count);
+                    outputStream.flush();
+                    progressBar.stepBy((long) NO_OF_BYTES);
+                }
+            }
+        } catch (IOException e) {
+            throw ErrorUtil.createCommandException("error occurred while uploading balo to central: " + e.getMessage());
+        }
+
+        try {
+            int statusCode = getStatusCode(conn);
+            // 200 - Module pushed successfully
+            // Other - Error occurred, json returned with the error message
+            if (statusCode == HttpURLConnection.HTTP_OK) {
+                outStream.println(org + "/" + name + ":" + version + " pushed to central successfully");
+            } else if (statusCode == HttpURLConnection.HTTP_UNAUTHORIZED) {
+                errStream.println("unauthorized access token for organization: " + org);
+            } else if (statusCode == HttpURLConnection.HTTP_BAD_REQUEST) {
+                try (BufferedReader reader = new BufferedReader(
+                        new InputStreamReader(conn.getErrorStream(), Charset.defaultCharset()))) {
+                    Error errorJsonSchema = new Gson().fromJson(reader, Error.class);
+
+                    if (errorJsonSchema.getMessage() != null && !"".equals(errorJsonSchema.getMessage())) {
+                        throw ErrorUtil.createCommandException(errorJsonSchema.getMessage());
+                    } else {
+                        throw ErrorUtil.createCommandException(
+                                ERR_CANNOT_PUSH + "'" + org + "/" + name + ":" + version + "' reason:" + reader.lines()
+                                        .collect(Collectors.joining("\n")));
+                    }
+                } catch (IOException e) {
+                    throw ErrorUtil.createCommandException(
+                            ERR_CANNOT_PUSH + "'" + org + "/" + name + ":" + version + "' to the remote repository '"
+                                    + conn.getURL() + "'");
+                }
+            } else {
```

Comment:
Push can also return `500` status code as well. Better to handle that once as well.

Reply:
This else part is handling all other statuses, Do we need to handle it separately? Since we are not getting any error message do we need to add another if condition?